### PR TITLE
Centos 7: use systemd socket activation instead of xinetd.

### PR DESCRIPTION
**Pull request by afics**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -25,7 +25,13 @@
     }
     'RedHat': {
       $package          = 'tftp-server'
-      $daemon           = false
+      if $::operatingsystemrelease =~ /^7/ {
```
Comment (ekohl) [inline manifests/params.pp]: Maybe EL6 is the exception here? I'd expect Fedora to have the tftp.socket service as well.

File: manifests/params.pp [hunk 2]
```diff
@@ -25,7 +25,13 @@
...
+        $daemon           = true
+        $service          = 'tftp.socket'
+        $service_provider = 'systemd'
```
Comment (ekohl) [inline manifests/params.pp]: Isn't this what Puppet does by default on EL7 (and Fedora)?

File: manifests/params.pp (hunk 1 above)
Comment (ekohl) [inline manifests/params.pp]: And I'd prefer versioncmp over a regex (even though we do the same below).

**Comment by ekohl** [comment]
I like this idea

File: manifests/params.pp (hunk 1 above)
Comment (afics) [inline manifests/params.pp]: I think you're right about EL6. Fedora 23 and 24 both have the `tftp.socket`.
Should I change it?

File: manifests/params.pp (hunk 2 above)
Comment (afics) [inline manifests/params.pp]: That I don't know.

File: manifests/params.pp (hunk 1 above)
Comment (ekohl) [inline manifests/params.pp]: I think ```if versioncmp($::operatingsystemrelease, '7') >= 0 {``` should work

File: manifests/params.pp (hunk 2 above)
Comment (ekohl) [inline manifests/params.pp]: You could test it by setting it to undef.

Comment (afics) [inline manifests/params.pp]: Does not work without explicitly setting it like above.

**Comment by ekohl** [comment]
Can you fix the tests as well?

**Comment by domcleal** [comment]
I was about to say, tests are failing - otherwise the change works OK.

The only problem is that xinetd isn't going to be unconfigured, but that could perhaps be addressed separately by retaining `xinetd::service` and setting its `ensure` property to `absent`, when daemon is true.

**Comment by afics** [comment]
I changed this, because I wanted to get rid of the xinetd class. Can we somehow achieve that while still unconfiguring xinetd, if it was in use previously?

As for the tests, I have never written rspec tests before, but I can look into it.

**Comment by domcleal** [comment]
> I changed this, because I wanted to get rid of the xinetd class. Can we somehow achieve that while still unconfiguring xinetd, if it was in use previously?

No, probably not. It may be better not to unconfigure it or even to use a separate parameter to remove the xinetd class.

**Comment by ekohl** [comment]
Can we help to move this forward?

**Comment by ekohl** [comment]
Maybe at first we could introduce it as a separate parameter to allow this, but not enable it by default so we our users have to deal with the migration instead of us?

**Comment by ekohl** [comment]
#59 has been merged. That sets the service so you can opt in by setting ```daemon``` to ```true``` on EL7 or Fedora. It doesn't do so by default to stay compatible and avoid a migration but I now consider this fixed.

Thansk for pointing out this was possible!

**Pull request closed without merging**